src/test/java/it/com/atlassian/bitbucket/jenkins/internal/config/BitbucketPluginConfigurationIT.java

```diff
@@ -25,17 +30,23 @@
     @ClassRule
     public static final BitbucketJenkinsRule bbJenkinsRule = new BitbucketJenkinsRule();
 
+    @Rule
+    public BitbucketJenkinsWebClientRule webClientRule = new BitbucketJenkinsWebClientRule(bbJenkinsRule);
+    @Rule
+    public BitbucketJenkinsLoggerRule bitbucketJenkinsLoggerRule = new BitbucketJenkinsLoggerRule();
```

Comment:
Can we somehow make this part of base Jenkins Rule? I guess the order of chaining is the problem. Then I think we should have a static factory inside BitbucketJenkinsRule to default wrap it in logger. Another static factory will simply return Jenkins rule without wrapping. The reason being, the chaining done in this class is more of an exception than the norm.

Reply:
`BitbucketJenkinsLoggerRule` extends `TestWatcher` to copy the logs over on failing tests only. `BitbucketJenkinsRule` already extends `JenkinsRule` so we can't get it to extend `TestWatcher` too